state/protocol/protocol_state/state/protocol_state.go

```diff
@@ -16,33 +16,65 @@ import (
 // on a per-block and per-epoch basis.
 // It is backed by a storage.ProtocolState and an in-memory protocol.GlobalParams.
 type ProtocolState struct {
```

Comment:
repeating myself: `ProtocolState.AtBlockID(id).Entry()` returns the configuraton at the beginning of the epoch (in accordance with the documentation), yet easily mistaken to be the entry at the specified block 😨

Reply:
Same as in other places, it's actually entry at the specific block since it was created using `AtBlockID`, there is no way to create "epoch" entry

Reply:
This is a problem with a confusing API but not with the business logic. Therefore my concern here was unwarranted -- see [discussion above](https://github.com/onflow/flow-go/pull/5650#discussion_r1562513689)